js/civicase-settings-form.js

```diff
@@ -16,6 +21,17 @@
         .each(showHideRelatedFormUrlField);
     }
 
+    /**
+     * Toggles the visibility of webforms button label text box
+     */
+    function showWebformsDropdownButtonLabel () {
+      var isWebformsDropdownVisible = parseInt($showWebforms.filter(':checked').val());
+
+      isWebformsDropdownVisible
+        ? $webformButtonLabel.parents('tr').show()
+        : $webformButtonLabel.parents('tr').hide();
+    }
```

Comment:
We are repeating the same functionality in `showHideRelatedFormUrlField`. Can we make both of them use generic classes for showing and hiding a companion field?

Reply:
Sorry @deb1990 this is not what I meant.

I meant creating a class that you add to the yes/no radio buttons which in turn would hide or show a companion input.

Like, if you'd configure the field's HTML attributes like this (as a very crude example):

```php
'title' => 'Show Webforms list in a separate dropdown',
'html_attributes' => [
  'data-toggles-visibility-for' => 'civicase__settings__webform-button-label',
  'class' => 'civicase__settings__visibility-toggler',
],

// ...

'title' => 'Label for the Webforms dropdown button',
'html_attributes' => [
  'class' => 'civicase__settings__webform-button-label',
],

```

If we add these classes and attributes to these elements + the webform URL elements (defined here https://github.com/compucorp/uk.co.compucorp.civicase/commit/de04a5e18cc77b705419893bcb75b7704117c10b#diff-827f957d95878344873e43d0ee49df39R44) then we'd have a solution that will work for both of them, plus any new fields that want to use a similar functionality.

If you have already spent too long of this you can leave as is. We can keep this in mind next time we touch the form.

Reply:
Nice idea @reneolivo . Done